assets/blueprints/genesis_helper/src/lib.rs

```diff
@@ -217,31 +243,71 @@ mod genesis_helper {
                         divisibility: 18,
                         metadata,
                         access_rules,
-                        initial_supply,
-                        resource_address: resource.address_bytes,
+                        initial_supply: resource.initial_supply,
+                        resource_address: address_bytes,
                     },
                 )
                 .unwrap(),
             );
 
-            for (account_address, amount) in initial_allocation {
-                let allocation_bucket = bucket.take(amount);
+            (resource_address, initial_supply_bucket)
+        }
+
+        fn allocate_resources(
+            &mut self,
+            accounts: Vec<ComponentAddress>,
+            allocations: BTreeMap<ResourceAddress, Vec<GenesisResourceAllocation>>,
+        ) {
+            for (resource_address, allocations) in allocations.into_iter() {
+                let mut resource_vault = self.resource_vaults.get_mut(&resource_address).unwrap();
+                for GenesisResourceAllocation {
+                    account_index,
+                    amount,
+                } in allocations.into_iter()
+                {
+                    let account_address = accounts[account_index as usize].clone();
+                    let allocation_bucket = resource_vault.take(amount);
+                    let _: () = Runtime::call_method(
+                        account_address,
+                        "deposit",
+                        scrypto_encode(&AccountDepositInput {
+                            bucket: allocation_bucket,
+                        })
+                        .unwrap(),
+                    );
+                }
+            }
+        }
+
+        fn allocate_xrd(&mut self, allocations: BTreeMap<ComponentAddress, Decimal>) {
+            for (account_address, amount) in allocations.into_iter() {
+                let bucket = self.xrd_vault.take(amount);
                 let _: () = Runtime::call_method(
                     account_address,
                     "deposit",
-                    scrypto_encode(&AccountDepositInput {
-                        bucket: allocation_bucket,
-                    })
-                    .unwrap(),
+                    scrypto_encode(&AccountDepositInput { bucket }).unwrap(),
                 );
             }
-            bucket.drop_empty();
+        }
+
+        pub fn wrap_up(&mut self) -> Bucket {
+            // A little hack to move to the next epoch,
+            // which also updates the validator set
+            // TODO: clean this up (add a dedicated method in epoch manager?)
```

Comment:
Yes, epoch_manager now has a method called `start` which may be the right method to start the epoch_manager